v2/streaming-data-generator/src/test/java/com/google/cloud/teleport/v2/templates/StreamingDataGeneratorIT.java

```diff
@@ -94,4 +128,171 @@ public void testFakeMessagesToGcs() throws IOException {
+        pubsubResourceManager.createSubscription(backlogTopic, "output-subscription");
+    // Arrange
+    String name = testName.getMethodName();
+    String jobName = createJobName(name);
+    LaunchConfig.Builder options =
+        LaunchConfig.builder(jobName, specPath)
+            .addParameter(SCHEMA_TEMPLATE_KEY, SchemaTemplate.GAME_EVENT.name())
+            .addParameter(QPS_KEY, DEFAULT_QPS)
+            .addParameter(SINK_TYPE_KEY, SinkType.PUBSUB.name())
+            .addParameter(TOPIC_KEY, backlogTopic.toString());
+
+    // Act
+    JobInfo info = launchTemplate(options);
+    assertThat(info.state()).isIn(JobState.ACTIVE_STATES);
+    Result result =
+        new DataflowOperator(getDataflowClient())
+            .waitForConditionAndFinish(
+                createConfig(info),
+                () -> pubsubResourceManager.pull(subscription, 5).getReceivedMessagesCount() > 0);
+    // Assert
+    assertThat(result).isEqualTo(Result.CONDITION_MET);
+  }
+
+  @Test
+  public void testFakeMessagesToBigQuery() throws IOException {
+    // Set up resource manager
+    bigQueryResourceManager =
+        DefaultBigQueryResourceManager.builder(testName.getMethodName(), PROJECT)
+            .setCredentials(credentials)
+            .build();
+    // schema should match schema supplied to generate fake records.
+    Schema schema =
+        Schema.of(
+            Field.of("eventId", StandardSQLTypeName.STRING),
+            Field.of("eventTimestamp", StandardSQLTypeName.INT64),
+            Field.of("ipv4", StandardSQLTypeName.STRING),
+            Field.of("ipv6", StandardSQLTypeName.STRING),
+            Field.of("country", StandardSQLTypeName.STRING),
+            Field.of("username", StandardSQLTypeName.STRING),
+            Field.of("quest", StandardSQLTypeName.STRING),
+            Field.of("score", StandardSQLTypeName.INT64),
+            Field.of("completed", StandardSQLTypeName.BOOL));
+    // Arrange
+    String name = testName.getMethodName();
+    String jobName = createJobName(name);
+    bigQueryResourceManager.createTable(jobName, schema);
+    LaunchConfig.Builder options =
+        LaunchConfig.builder(jobName, specPath)
+            .addParameter("schemaTemplate", String.valueOf(SchemaTemplate.GAME_EVENT))
+            .addParameter("qps", "1000000")
+            .addParameter("sinkType", "BIGQUERY")
+            .addParameter(
+                "outputTableSpec",
+                String.format(
+                    "%s:%s.%s", PROJECT, bigQueryResourceManager.getDatasetId(), jobName));
+
+    // Act
+    JobInfo info = launchTemplate(options);
+    assertThat(info.state()).isIn(JobState.ACTIVE_STATES);
+    Result result =
+        new DataflowOperator(getDataflowClient())
+            .waitForConditionAndFinish(
+                createConfig(info),
+                () -> bigQueryResourceManager.readTable(jobName).getTotalRows() > 0);
+    // Assert
+    assertThat(result).isEqualTo(Result.CONDITION_MET);
```

Comment:
This is hard to do since the generated record is fake and created using a very vague schema template. I think the fact that the data exists in the BQ/Spanner table proves the validity of the data (since it has to match the schema of the table)